### ISSUE TITLE
Show freeze stars instead of bar in DDRace servers

### ISSUE THREAD

**Issue by Kruimeldief** [ddnet/ddnet#6028]
**Edit** correct: show freeze stars instead of bar (without the old shield + heart HUD).

Can the health and ammo HUD return to DDRace servers (similar to v16.1)? The freeze duration shield countdown feels much more precise to me than the new freeze bar. The setting checkbox can be added to `settings => appearance => HUD => DDRace HUD`. The image below shows the empty spot for the `show health + ammo` option when you deselect the `show DDRace HUD` checkbox.

![DDNet HUD appearance settings](https://i.imgur.com/pcBJYIc.gif)

**Comment by fokkonaut** [comment]
No. But you can enable this on a modded server on the server-side.

**Comment by edg-l** [comment]
not needed

**Comment by heinrich5991** [comment]
I don't think we're going to reintroduce shields. What can we do to improve the freeze bar accuracy for you?

**Comment by Kruimeldief** [comment]
Correction: it's not the shields but the damage stars around the tee that helps me predict when the tee unfreezes. Would it be acceptable to return the stars without the heart + shield HUD? To me, the stars look nicer too, but I know that's a matter of taste.

> What can we do to improve the freeze bar accuracy for you?
Maybe I need to play more with the new bar. But currently (in v16.1), I use the rhythm of the start (3-2-1 stars) to predict when I can move again. It's kind of like a metronome countdown. When I use the bar, it's difficult to distinguish the last few pixels of the bar. And because the bar timer goes down smoothly, there's no rhythm (feeling/prediction) to hold onto.

Below are two suggestions to make the freeze bar more accurate in terms of predicting when you will be unfrozen. But I know they are not ideal, which is why I prefer the stars.
- Increase the length of the bar (by 2 or 3 times).
- Make the bar logarithmic: starts slow and ends faster.

**Comment by Kruimeldief** [comment]
Correction: it's not the shields but the damage stars around the tee that helps me predict when the tee unfreezes. Would it be acceptable to return the stars without the heart + shield HUD? To me, the stars look nicer too, but I know that's a matter of taste.

> I don't think we're going to reintroduce shields. What can we do to improve the freeze bar accuracy for you?

Maybe I need to play more with the new bar. But currently (in v16.1), I use the rhythm of the start (3-2-1 stars) to predict when I can move again. It's kind of like a metronome countdown. When I use the bar, it's difficult to distinguish the last few pixels of the bar. And because the bar timer goes down smoothly, there's no rhythm (feeling/prediction) to hold onto.

Below are two suggestions to make the freeze bar more accurate in terms of predicting when you will be unfrozen. But I know they are not ideal, which is why I prefer the stars.
* Increase the length of the bar (by 2 or 3 times).
* Make the bar logarithmic: starts slow and ends faster.

**Comment by heinrich5991** [comment]
Perhaps it would help to make the bar flash on each whole second?

**Comment by Chairn** [comment]
> Perhaps it would help to make the bar flash on each whole second?

Please don't. However, adding some marks in the freeze bar may help?

**Comment by Learath2** [comment]
I am open to adding an alternate indicator client-side that mimics the stars. It shouldn't be too much of a maintainence hassle, I just haven't had the time to implement it.

**Comment by ErrorDreemurr** [comment]
I think the CNetObj_DDNetCharacter should can toggle HUD.
Because if you create a PVP mod based on DDNet.
You want to show freeze bar, so must snap CNetObj_DDNetCharacter.
But if you snap CNetObj_DDNetCharacter, you can't see the health, ammo and armor.
So this is a problem.

**Comment by Learath2** [comment]
I think you can snap ddnetcharacter and show the hud just fine using the gameinfo flags `HUD_*`. @fokkonaut does it with his mod iirc

**Comment by WatermelonDriveway** [comment]
> I use the rhythm of the start (3-2-1 stars) to predict when I can move again. It's kind of like a metronome countdown. When I use the bar, it's difficult to distinguish the last few pixels of the bar.

This is very true, and exactly the reason I was surprised when the stars got removed.  I've seen a few techniques that rely on the moment a tee unfreezes.

![Screen Shot 2022-11-15 at 12 55 07 AM](https://user-images.githubusercontent.com/8306685/201850318-4a5b890d-f7b5-4007-8217-a8b6993e8b73.png)

Here you want to hook the bottom tee just when they unfreeze.  They will be holding right while frozen, so if you're too early with your hook, it will expire and they will fall, if you're too late, they will have moved right already and your hook will miss.  It's absolutely possible to get close enough timing with the freeze bar, but the lack of a countdown removes a level of accuracy that is super helpful for parts like this.  When I first ran this map I died here several times before realizing I could use the stars as a timing reference.

**Comment by Robyt3** [comment]
How about adding indicator lines for every second to the freeze bar?

![screenshot_2023-02-02_00-07-15](https://user-images.githubusercontent.com/23437060/216188608-160f41b1-573a-421b-951e-802d321dfb18.png)

![screenshot_2023-02-02_00-07-41](https://user-images.githubusercontent.com/23437060/216188614-bf14c7c2-705b-4332-a7f5-f1ced80e3fa4.png)

![screenshot_2023-02-02_00-07-52](https://user-images.githubusercontent.com/23437060/216188613-8a1b1f97-883c-4a96-a6f7-ddd47a52a799.png)

Maximum of 4 indicator lines because it's not very useful to have them if the freeze time is large.

**Comment by def-** [comment]
I like it, not intrusive

**Comment by Kruimeldief** [comment]
I understand that you may want to implement this feature for others but it won't help my needs. But to be honest, I don't mind it. I'm still using the latest version that has the old ninja stars and I'm fine with that.

The reason why I think it's not helpful is shown in the last picture. In my opinion, the stars are all about predicting (through the rhythm) when you'll be unfrozen. Adding indicated at the far left when you're frozen for 10+ seconds doesn't help. It may help when you're frozen the traditional 3 seconds because you can make a rhythm yourself by looking when the blue filler crosses the indicator lines. But even then, I feel like I'm actually closely inspecting the moment when the bar runs empty.

So in short, it's a nice addition to DDNet and you'll probably make players happy with it. But it won't solve my issue. I hope older versions will stay compatible with the servers.

**Comment by fokkonaut** [comment]
I want the stars (or something similar @Learath2) back aswell. Also the fact, they lasted for exactly one second, made them perfect for timing! Bars are nothing against that!

**Comment by rockuSSS** [comment]
DDNet has never been the same since stars were removed, add them back!

**Comment by Soraah** [comment]
Give stars back =P Also a option to disable the idle sitting animation :)

**Comment by RemakePower** [comment]
> I think you can snap ddnetcharacter and show the hud just fine using the gameinfo flags `HUD_*`. @fokkonaut does it with his mod iirc

Yes, thanks. When i comment last year, i don't find this.

**Comment by Robyt3** [comment]
#8934 added an option to reenable the stars.
